tools/dockerfiles/dockerfile_gcc_540

```diff
@@ -0,0 +1,82 @@
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+# GCC 5.2.0 Cross Compiler Dockerfile
+#
+# This dockerfile creates a GCC cross compiler for "x86_64-elf" that can be
+# used to compile source code on the host system. Note that this cross
+# compiler is intended to be used with the bareflank-gcc-wrapper that brings
+# in the libc/libc++ sysroot.
+#
+# To give this docker image the ability to compile code, we overlay the
+# host's source files into the docker container with the same path. This
+# allows the host to use it's paths untouched and still use the cross compiler
+# in docker to compile code.
+#
+# To install docker on Ubuntu, use the following:
+#
+# - sudo apt-get install docker.io
+# - sudo usermod -a -G docker $USER
+#
+# To create this docker image:
+#
+# - cd <hypervisor repo>
+# - docker build -t bareflank/gcc_540 -f tools/dockerfiles/dockerfile_gcc_540 .
+
+FROM ubuntu:16.04
+
+# ------------------------------------------------------------------------------
+# Packages
+# ------------------------------------------------------------------------------
+
+RUN apt-get update -y
+
+RUN apt-get install -y build-essential
+RUN apt-get install -y libgmp-dev
+RUN apt-get install -y libmpc-dev
+RUN apt-get install -y libmpfr-dev
+RUN apt-get install -y flex
+RUN apt-get install -y bison
+RUN apt-get install -y wget
+RUN apt-get install -y texinfo
+RUN apt-get install -y git
+RUN apt-get install -y cmake
+
+# ------------------------------------------------------------------------------
+# Files
+# ------------------------------------------------------------------------------
+
+RUN mkdir -p /tools/scripts/
+
+COPY ./tools/scripts/create-cross-compiler.sh /tools/scripts/create-cross-compiler.sh
+
+COPY ./tools/scripts/fetch_binutils.sh /tools/scripts/fetch_binutils.sh
+COPY ./tools/scripts/fetch_gcc.sh /tools/scripts/fetch_gcc.sh
+COPY ./tools/scripts/fetch_nasm.sh /tools/scripts/fetch_nasm.sh
+
+COPY ./tools/scripts/build_binutils.sh /tools/scripts/build_binutils.sh
+COPY ./tools/scripts/build_gcc.sh /tools/scripts/build_gcc.sh
+COPY ./tools/scripts/build_nasm.sh /tools/scripts/build_nasm.sh
+
+# ------------------------------------------------------------------------------
+# Script
+# ------------------------------------------------------------------------------
+
+RUN CROSS_COMPILER=gcc_540 ./tools/scripts/create-cross-compiler.sh
```

Comment:
Usually you add a VOLUME and WORKDIR to the Dockerfiles to ensure there is a partition there like /bareflank and it will automatically run in that directory if you add: 
VOLUME /bareflank
WORKDIR /bareflank

This reduces the worry that when you use the docker file in a script a docker update might change the default working dir if  you mount stuff into the container.